firehose/parsers/flawfinder.py

```diff
@@ -0,0 +1,117 @@
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+#   USA
+
+import sys
+import re
+from subprocess import check_output
+from firehose.model import Message, Point, \
+    File, Location, Generator, Metadata, Analysis, Issue
+
+
+def main():
+    """ Main entry to flawfinder parser """
+    try:
+        arg_file = sys.argv[1]
+        report = open(arg_file, 'r')
+    except IOError as e:
+        raise e
+    except IndexError:
+        print("Missing input file")
+    analysis = parse_file(report)
+    sys.stdout.write(str(analysis.to_xml()))
+    sys.stdout.write('\n')
+
+
+def parse_file(infile):
+    """ Parser flawfinder output
+
+    :infile: file-like object
+    :returns: Firehose Analysis object, representing the final XML.
+
+    Flawfinder can generate multiple cwes for a single issue.
+    Firehose's models does not supports multiple CWEs.
+    For now, when  multiple CWEs ocurrs, we get only the first one.
+
+    A issue was created to track this bug:
+    https://github.com/fedora-static-analysis/firehose/issues/35
+    """
+
+    line = infile.readline()
+    generator = Generator(name='flawfinder',
+                          version=get_flawfinder_version(line))
+    metadata = Metadata(generator, None, None, None)
+    analysis = Analysis(metadata, [])
+
+    pattern = "\([a-z]*\)"
+    pattern2 = "\[[0-9]\]"
+    prog = re.compile(pattern)
+    prog2 = re.compile(pattern2)
+
+
+    while line:
+        if line != "\n" and (prog.search(line) and prog2.search(line)):
+
+            testid = re.search("\(([a-z]+)\)", line).group(1)
+            issue_path = line.split(":")[0]
+            issue_line = line.split(":")[1]
+
+            location = Location(file=File(issue_path, None),
+                                function=None,
+                                point=Point(int(issue_line), 0))
+
+            message_line = infile.readline()
+            issue_message = ""
+            cwes = []
+            while message_line != "\n" and not \
+            (prog.search(message_line) and prog2.search(message_line)):
+                issue_message += " " + message_line
+                cwes.extend([int(cwe) for cwe in re.findall("CWE-([0-9]+)", issue_message)])
+                message_line = infile.readline()
+            line = message_line
+
+            if cwes:
+                first_cwe = int(cwes[0])
+            else:
+                first_cwe = None
+
+            issue = Issue(first_cwe, testid, location,
+                          Message(text=issue_message), notes=None,
+                          trace=None, severity=None, customfields=None)
```

Comment:
Please set the ``severity`` field to the number in square parens (keep it as a string, rather than calling ``int`` on it).